crates/saya/core/src/dojo_os/piltover.rs

```diff
@@ -0,0 +1,47 @@
+use anyhow::Context;
+use cairo_proof_parser::to_felts;
+use dojo_world::migration::TxnConfig;
+use dojo_world::utils::TransactionExt;
+use serde::Serialize;
+use starknet::accounts::{Account, Call};
+use starknet::core::utils::get_selector_from_name;
+use starknet_crypto::Felt;
+use tracing::trace;
+
+use crate::verifier::utils::wait_for_sent_transaction;
+use crate::{SayaStarknetAccount, LOG_TARGET};
+
+#[derive(Debug, Serialize)]
+pub struct PiltoverCalldata {
```

Comment:
I'm not sure if the cainome bindgen is already integrated to piltover. But if yes, we should use it to already have concrete types in rust that are automatically serializable in cairo.

Reply:
Looks to me like it is integrated, will use it as a dependcy.

By the way those are automatically serializable, as they use `serde_felt` from this PR: https://github.com/cartridge-gg/cairo-proof-parser/pull/10.

Reply:
It doesn't seem to be working at the moment. Create a bug report here: https://github.com/keep-starknet-strange/piltover/issues/39

Reply:
Thanks for that, I'll have a look at it to ensure it works as it will be better to rely on generated code.

Are we using a fork of piltover or the original contract?

Reply:
I'm using this one, [https://github.com/matzayonc/piltover/tree/feat/fact-registry](https://github.com/matzayonc/piltover/tree/feat/fact-registry).